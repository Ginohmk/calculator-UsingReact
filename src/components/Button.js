import React from 'react';
import PropTypes from 'prop-types';

function Button(props) {
  const { title } = props;
  const { className } = props;
  const { perform } = props;

  return (
    <button
      onClick={perform}
      className={className}
      data-testid={title}
      type="button"
    >
      {title}
    </button>
  );
}

Button.propTypes = {
  title: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  perform: PropTypes.func.isRequired,
};
export default Button;
