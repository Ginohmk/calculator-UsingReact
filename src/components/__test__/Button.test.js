import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('TEst for Home component', () => {
  test('Home snapShot test', () => {
    const tree = renderer
      .create(
        <Button perform={() => test('AC')} title="AC" className="operation" />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
