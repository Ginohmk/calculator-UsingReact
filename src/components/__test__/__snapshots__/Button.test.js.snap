// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEst for Home component Home snapShot test 1`] = `
<button
  className="operation"
  onClick={[Function]}
  type="button"
>
  AC
</button>
`;
